File: 프로그래머스/lv2/60057. 문자열 압축/문자열 압축.js
function solution(s) {
    var answer = 0;
    let result = []
//     매개변수 서치? 아 근데, 일방향이 아닌데....
//     그냥 완탐?
    for(let i=1;i<=s.length;i++){
        let tmp = ""
        let tmp_res = []
        let index = -1
        for(let j=0;j<=s.length;j+=i){
            let tmp_slice = s.slice(j,j+i)
            if(tmp_slice){
                if(tmp === tmp_slice){
                    tmp_res[index][0]+=1
                }
                else{
                    index++
                    tmp_res.push([1,tmp_slice])
                    tmp = tmp_slice
                }
            }
        }
        let count = 0
        for(let x of tmp_res){
            [num,str] = x
            if(num!==1){
                count+=(str.length+String(num).length)
            }
            else{
                count+=str.length
            }
        }
        result.push(count)
    }
    answer = Math.min(...result)
    
    return answer;
}

File: 프로그래머스/lv2/60057. 문자열 압축/문자열 압축.py
def solution(s):
    answer = len(s)
#     맨앞에서 잘라야한다는게 함정
    for i in range(1, len(s) // 2 + 1):
        tmp = ""
        count = 1
        length =0
        for u in range(0,len(s)+1,i):
            if s[u:u+i] == tmp:
                count+=1
            else:
                length +=len(s[u:u+i])
                if count>=2:
#                     숫자길이더하기
                    length += len(str(count))
                count = 1
                tmp = s[u:u+i]
        answer = min(answer,length)
    return answer